core/src/test/java/org/elasticsearch/search/aggregations/InternalAggregationTestCase.java

```diff
@@ -122,4 +135,56 @@ protected final T createUnmappedInstance(String name) {
     protected NamedWriteableRegistry getNamedWriteableRegistry() {
         return namedWriteableRegistry;
     }
+
+    protected ContextParser<Object, Aggregation> instanceParser() {
+        throw new UnsupportedOperationException("Not implemented yet");
+    }
```

Comment:
Oh ok, I see what you meant.

Yes, that's the way to go, I didn't think about overriding xContentRegistry() and that will help. I usd instanceParser() to mimic instanceReader() we already have, but you suggestion is better. I'll update the pull request.